coder-login/main.tf

```diff
@@ -0,0 +1,30 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.12"
+    }
+  }
+}
+
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+}
+
+data "coder_workspace" "me" {}
+
+resource "coder_script" "coder-login" {
+  agent_id = var.agent_id
+  script = templatefile("${path.module}/run.sh", {
+    CODER_USER_TOKEN : data.coder_workspace.me.owner_session_token,
+    CODER_DEPLOYMENT_URL : data.coder_workspace.me.access_url
+  })
+  display_name       = "Coder Login"
+  icon               = "../.icons/coder-white.svg"
```

Comment:
This is run on the Coder server, so it cannot access the .icons folder in the modules repo. 
You can use /icon/coder.svg after you upload the icon to 
https://github.com/coder/coder/tree/main/site/static/icon
or use full HTML path to the coder icon

Reply:
It works great now.